src/lib/Form/Type/Event/ContentCreateContentTypeChoiceLoaderEvent.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Ibexa\AdminUi\Form\Type\Event;
+
+use Symfony\Contracts\EventDispatcher\Event;
+
+final class ContentCreateContentTypeChoiceLoaderEvent extends Event
+{
+    public const RESOLVE_CONTENT_TYPES = 'admin_ui.content_create.content_type_resolve';
+
+    /** @var array<string, array<\Ibexa\Contracts\Core\Repository\Values\ContentType\ContentType>> */
+    private array $contentTypeGroups;
+
+    public function __construct(array $contentTypeGroups)
+    {
+        $this->contentTypeGroups = $contentTypeGroups;
+    }
+
+    /**
+     * @return array<string, array<\Ibexa\Contracts\Core\Repository\Values\ContentType\ContentType>>
+     */
+    public function getContentTypeGroups(): array
+    {
+        return $this->contentTypeGroups;
+    }
+
+    /**
+     * @param array<string, array<\Ibexa\Contracts\Core\Repository\Values\ContentType\ContentType>> $contentTypeGroups
+     */
+    public function setContentTypeGroups(array $contentTypeGroups): void
+    {
+        $this->contentTypeGroups = $contentTypeGroups;
+    }
```

Comment:
I would suggest adding a `addContentTypeGroup` (and `removeContentTypeGroup` possibly) event methods. Maybe even `addContentType($group, $contentType)`?

This will simplify the actual usage, where you will no longer be forced to merge the arrays. Instead you would only call said method.

Reply:
Yes, that is indeed the code I was looking at.

Merging of configuration arrays happens in the other EventSubscriber in the linked PR, but we are not changing that part. Yet? :sweat_smile:

Reply:
Done in https://github.com/ibexa/admin-ui/pull/869/commits/1f8358ef487870b6d0ac3be4e4129c8333ee705b